nixos/modules/programs/sway.nix

```diff
@@ -4,82 +4,30 @@ with lib;
 
 let
   cfg = config.programs.sway;
-  swayPackage = pkgs.sway;
-
-  swayWrapped = pkgs.writeShellScriptBin "sway" ''
-    if [[ "$#" -ge 1 ]]; then
-      exec sway-setcap "$@"
-    else
-      ${cfg.extraSessionCommands}
-      exec ${pkgs.dbus.dbus-launch} --exit-with-session sway-setcap
-    fi
-  '';
-  swayJoined = pkgs.symlinkJoin {
-    name = "sway-joined";
-    paths = [ swayWrapped swayPackage ];
-  };
+  swayPackage = cfg.package;
 in {
   options.programs.sway = {
     enable = mkEnableOption ''
-      the tiling Wayland compositor Sway. After adding yourself to the "sway"
-      group you can manually launch Sway by executing "sway" from a terminal.
-      If you call "sway" with any parameters the extraSessionCommands won't be
-      executed and Sway won't be launched with dbus-launch'';
-
-    extraSessionCommands = mkOption {
-      type = types.lines;
-      default = "";
-      example = ''
-        # Define a keymap (US QWERTY is the default)
```

Comment:
It might be nice to keep this since some users will need to set various XKB_ env vars before starting sway, but OTOH, if we remove it we can remove the sway wrapper entirely...

thoughts?

Reply:
We can remove this for `Sway 1.0` because it can/should be configured in the sway configuration, see https://github.com/swaywm/sway/wiki#keyboard-layout, but we should make that obvious (e.g. print an error message with instructions, if this option is set after the switch to 1.0).

Reply:
Oh great, I'd totally misread the docs and thought they were still env-var-based in 1.0. Thanks.